packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -367,18 +367,20 @@ class ScheduleManagerCore {
      * to make decision if op processing should be paused or not afer that.
      */
      public afterOpProcessing(sequenceNumber: number) {
-        // If the inbound queue is ever empty we pause it and wait for new events
-        if (this.deltaManager.inbound.length === 0) {
-            this.setPaused(true);
-            return;
-        }
+        assert(!this.localPaused, "can't have op processing paused if we are processing an op");
 
         // If no message has caused the pause flag to be set, or the next message up is not the one we need to pause at
         // then we simply continue processing
-        if (this.pauseSequenceNumber !== undefined && sequenceNumber + 1 >= this.pauseSequenceNumber) {
+        if (this.pauseSequenceNumber !== undefined) {
+            assert(sequenceNumber < this.pauseSequenceNumber, "processed op we should have not processed");
```

Comment:
I didn't quite follow these 2 cases here -
1. sequenceNumber < this.pauseSequenceNumber
2. sequenceNumber + 1 === this.pasueSequenceNumber

Could you add a comment explaining why this code is expecting this. For instance, when can this.pauseSequenceNumber be greater that sequenceNumber + 1? And why do we need to pause if they are the same.